src/WebJobs.Script.WebHost/Security/KeyManagement/SecretManager.cs

```diff
@@ -90,12 +89,26 @@ public async virtual Task<HostSecretsInfo> GetHostSecretsAsync()
                         _logger.LogDebug("Loading host secrets");
 
                         hostSecrets = await LoadSecretsAsync<HostSecrets>();
-                        if (hostSecrets == null)
+                        try
                         {
-                            // host secrets do not yet exist so generate them
-                            _logger.LogDebug(Resources.TraceHostSecretGeneration);
-                            hostSecrets = GenerateHostSecrets();
-                            await PersistSecretsAsync(hostSecrets);
+                            if (hostSecrets == null)
+                            {
+                                // host secrets do not yet exist so generate them
+                                _logger.LogDebug(Resources.TraceHostSecretGeneration);
+                                hostSecrets = GenerateHostSecrets();
+                                await PersistSecretsAsync(hostSecrets);
+                            }
+                        }
+                        catch (Exception ex)
+                        {
+                            _logger.LogDebug("Potential conflict, attempting to load host secrets for the second time.");
+                            hostSecrets = await LoadSecretsAsync<HostSecrets>();
+
+                            if (hostSecrets == null)
+                            {
+                                _logger.LogError(ex, "Failed to load host secrets on second attempt");
```

Comment:
Here... do *not* include the exception b/c it's not the one that actually caused the problem... and instead of `Failed to load...` maybe say `Host secrets are still null on second attempt.` We may have actually successfully loaded them, but there just weren't any...

Reply:
Changed!